modules/administration/pages/tshoot-iss.adoc

```diff
@@ -0,0 +1,56 @@
+[[troubleshooting-iss]]
+= Troubleshooting Inter-Server Synchronization
+
+////
+PUT THIS COMMENT AT THE TOP OF TROUBLESHOOTING SECTIONS
+
+Troubleshooting format:
+
+One sentence each:
+Cause: What created the problem?
+Consequence: What does the user see when this happens?
+Fix: What can the user do to fix this problem?
+Result: What happens after the user has completed the fix?
+
+If more detailed instructions are required, put them in a "Resolving" procedure:
+.Procedure: Resolving Widget Wobbles
+. First step
+. Another step
+. Last step
+////
+
+////
+Cause: Caches on the ISS slave and master can contain old or out-of-date entries.
+Consequence: Synchronizing fails, with ``ERROR: Encountered IntegrityError``.
+Fix: Delete the cache on the slave and the master
+Result: Synchronization completes successfully.
+////
+
+Caches on the ISS Master and Slave can contain old or out-of-date entries.
+This leads to synchronization failing with ``ERROR: Encountered IntegrityError: null value in column "installer_updates" violates not-null constraint``.
```

Comment:
Hmm, I would suggest to put this in a slightly different angle.

Story is: in the particular case of ISS, we make extensive use of caches, and those are particularly prone to be poisoned by bugs. That means that if version X has a bug, it might create invalid cache files - no surprise. What might be surprising is that even if user updates to version X+1 which has the bug fixed, the same symptoms might still show up. The reason is that cache files are not necessarily re-created every time, so "poisoned" caches from version X may well be still consumed in version X+1, resulting in the same symptoms even if the root cause had been fixed.

So the way to address this properly is not only to upgrade to X+1, but also to clear all caches so that we're sure to start from clean data.

The error message `null value in column "installer_updates" violates not-null constraint` is just the latest symptom of the latest bug of this kind. I would not focus on this message other than being an example at most. 

The following two lines are way more important, as they point directly to this (meta)problem:

```
consider removing satellite-sync cache at /var/cache/rhn/satsync/* and re-run satellite-sync with same options.
If this error persits after removing cache, please contact SUSE support.
```

Reply:
Updated.